### ISSUE TITLE
Juhani quest not finished after Juhani has been killed

### ISSUE THREAD

**Issue by OverDrone** [KOTORCommunityPatches/K1_Community_Patch#424]
If you side with DS Bastila and kill Juhani and Jolee Bindo, Juhani's personal quest "Juhani" (not "Threat from Xor") will never finish. Guides say that it should finish when you return to Ebon Hawk from temple. Can provide save file if needed. 
My  installation:
1. latest steam KOTOR
2. uniws
3. Latest community Patch v 1.8.1
4. Various HD mods (tga, txi, etc) that don't touch gameplay, quests, balance etc.
[save.zip](https://github.com/KOTORCommunityPatches/K1_Community_Patch/files/5177900/save.zip)

**Comment by JCarter426** [comment]
You can't finish it by the time you travel to the Star Forge system anyway. The quest should be cleared out before then, but perhaps it isn't.

**Comment by OverDrone** [comment]
> ...The quest should be cleared out...
Can you provide exact moment when this quest is cleared so I could confirm bug? Because I just got inside Star Forge, probably it's cleared a bit later. I couldn't find any info in inet guides when this quest should disappear during DS ending, only for LS: talk to Juhani on Ebon Hawk after temple.

**Comment by JCarter426** [comment]
All other unfinished quests are scrubbed by the script k_pebn_remove which I believe fires as soon as you land.

**Comment by OverDrone** [comment]
Then it is surely a bug, because attached save contains this quest and it's star forge after cutscene and jedi/sith fight.

**Comment by JCarter426** [comment]
Hmm, I took a deeper look and it seems the personal quests are not scrubbed, and I'm not sure if that's right or not. I don't believe there is any way of pursuing the quests in a way that makes any difference once you've landed. It looks like I got stuck with "maybe Jolee will talk to you later" as late as the Star Forge in one game when I went to Kashyyyk late. But if that's the way they did it.

It seems that in the light side ending, the quest state will be updated to 120 when after you defeat Bastila. mine went from 100 to 120 but there are possibly other states you could be in, if you were dark-sided but picked the light side ending, or completed the romance. It doesn't look like there is a journal entry for the dark side ending, so it seems that the simplest solution would be to remove the quest then.

Are you stuck with Jolee's as well?

**Comment by OverDrone** [comment]
No. I've finished Jolee quest just via conversation. How can I remove the quest? My version of KSE does not allow to edit quests in K1, only in J2. My main is male, so no romance. I've picked ds ending. Also I changed my alignment to ls with cheats. Will return to save and try without cheat. Do you know scripts consider pc alignment or only decisions? Can it be the cause?

**Comment by JCarter426** [comment]
It should be possible to remove it with KSE but if you are unable to do that, then there is no way for you to remove it. It's likely an oversight.

**Comment by DarthParametric** [comment]
We could always add some new DS end states to both the Jolee (`k_swg_joleetalk`) and Juhani (`k_swg_juhani`) quests, like the sort of thing we did for Dustil. 

But I notice that Juhani's final quest state (115) for a female PC isn't flagged as a quest end like it presumably should be (and unlike Jolee, who has two end states): "_Juhani has thanked you for everything you have done for her, and has expressed her feelings for you. You seem to mean more to her than she could ever admit before. Perhaps, once this is all over, there may be something for you._"

And I assume 105 is as far as a male PC gets: "_Juhani doesn't think that the path you tread now is the same one you followed when your armies freed her. She hopes that your actions will prove yourself in the future._"

**Comment by JCarter426** [comment]
Yeah, but it looks like most of the other characters aren't given dark side journal entries either. It looks like "Mission's Brother" is simply deleted and I did get stuck with Carth and Jolee's in the completed quests for my last dark sided game, despite the last entry not making sense with them gone anymore.

I think deleting it would be fine, unless you want to add them for all of them. It may be a case of BioWare assuming most players would want to be Jedi and not giving the dark side ending priority.

**Comment by DarthParametric** [comment]
It wouldn't be anything fancy, just a line or two about killing them after they stood against you on the Temple roof. But it's a good point that Carth should get similar treatment of the beach if he also lacks one (I think usually I'm at 99 with him there so not sure I ever noticed). Does Mission lack one as well? I always do Griff's stuff, so she wouldn't normally have any active quest for me.

**Comment by JCarter426** [comment]
Yeah, that sounds fine by me.

I'm pretty sure I had done Mission's quest but it was completely gone from my save, even though the others were still there as completed quests. Maybe they treated it differently because the quest is "Mission's Brother" whereas the others are "Carth" and "Jolee" etc.

**Comment by OverDrone** [comment]
Maybe just remove it from quest list same as LS? I don't think somebody would expect something more because in LS ending the quest is just removed. Ofc you may do as you described, add some DS journal entry, but I suspect it's a bit more coding effort than just add quest removal.
